internal/exec/exec.go

```diff
@@ -0,0 +1,487 @@
+	e.attrList, err = winapi.NewProcThreadAttributeList(3)
+	if err != nil {
+		return fmt.Errorf("failed to initialize process thread attribute list: %w", err)
+	}
+	siEx.ProcThreadAttributeList = e.attrList
+
+	// Need to know whether the process needs to inherit stdio handles.
+	inheritHandles := e.stdioProcEnd[0] != nil || e.stdioProcEnd[1] != nil || e.stdioProcEnd[2] != nil
+	if inheritHandles {
+		var handles []uintptr
+		for _, file := range e.stdioProcEnd {
+			if file.Fd() != uintptr(syscall.InvalidHandle) {
+				handles = append(handles, file.Fd())
+			}
+		}
+
+		// Set up the process to only inherit stdio handles and nothing else.
+		err = winapi.UpdateProcThreadAttribute(
+			siEx.ProcThreadAttributeList,
+			0,
+			windows.PROC_THREAD_ATTRIBUTE_HANDLE_LIST,
+			unsafe.Pointer(&handles[0]),
+			uintptr(len(handles))*unsafe.Sizeof(handles[0]),
+			nil,
+			nil,
+		)
+		if err != nil {
+			return err
+		}
+
+		// Assign the handles to the startupinfos stdio fields.
+		if e.stdioProcEnd[0] != nil {
+			siEx.StdInput = windows.Handle(e.stdioProcEnd[0].Fd())
+		}
+		if e.stdioProcEnd[1] != nil {
+			siEx.StdOutput = windows.Handle(e.stdioProcEnd[1].Fd())
+		}
+		if e.stdioProcEnd[2] != nil {
+			siEx.StdErr = windows.Handle(e.stdioProcEnd[2].Fd())
+		}
+	}
+
+	// Update the attribute list for pseudo console and job object setup if requested.
+
+	// if e.conPty != nil {
+	// 	if err := e.conPty.UpdateProcThreadAttribute(siEx.ProcThreadAttributeList); err != nil {
+	// 		return err
+	// 	}
+	// }
+
+	if e.job != nil {
+		if err := e.job.UpdateProcThreadAttribute(siEx.ProcThreadAttributeList); err != nil {
+			return err
+		}
+	}
+
+	var zeroSec windows.SecurityAttributes
+	pSec := &windows.SecurityAttributes{Length: uint32(unsafe.Sizeof(zeroSec)), InheritHandle: 1}
+	tSec := &windows.SecurityAttributes{Length: uint32(unsafe.Sizeof(zeroSec)), InheritHandle: 1}
+
+	siEx.Cb = uint32(unsafe.Sizeof(*siEx))
+	if e.execConfig.token != 0 {
+		err = winapi.CreateProcessAsUser(
+			e.execConfig.token,
+			argv0p,
+			argvp,
+			pSec,
+			tSec,
+			inheritHandles,
+			flags,
+			createEnvBlock(addCriticalEnv(dedupEnvCase(true, e.env))),
+			dirp,
+			&siEx.StartupInfo,
+			pi,
+		)
+	} else {
+		err = windows.CreateProcess(
+			argv0p,
+			argvp,
+			pSec,
+			tSec,
+			inheritHandles,
+			flags,
+			createEnvBlock(addCriticalEnv(dedupEnvCase(true, e.env))),
+			dirp,
+			&siEx.StartupInfo,
+			pi,
+		)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to create process: %w", err)
+	}
+	// Don't need the thread handle for anything.
+	defer windows.CloseHandle(windows.Handle(pi.Thread)) //nolint:errcheck
+
+	// Grab an *os.Process to avoid reinventing the wheel here. The stdlib has great logic around waiting, exit code status/cleanup after a
+	// process has been launched.
+	e.process, err = os.FindProcess(int(pi.ProcessId))
+	if err != nil {
+		return fmt.Errorf("failed to find process after starting: %w", err)
```

Comment:
is it possible to cancel/delete the process here using the handle, since, if it cant be found, none of the operations below will work on it (ie, it can't be monitored or killed)

Reply:
Added logic to terminate the process if os.FindProcess fails